### PR TITLE
virtio-serial:Under named-pipe non-blocking testing

### PR DESCRIPTION
**Pull request by nanliu-r**
id: 1848293

Signed-off-by: Nana Liu <nanliu@redhat.com>

depends on: [avocado-framework/avocado-vt#2400](https://github.com/avocado-framework/avocado-vt/pull/2400)

### CODE REVIEW

**Comment by nanliu-r** [comment]
@yanan-fu please add comments in this patch, thanks.

**Comment by nanliu-r** [comment]
Test pass:
 (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.virtio_trace_pipenb: PASS (73.86 s)

File: qemu/tests/virtio_trace_pipenb.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    """
+    timeout = float(params.get("login_timeout", 360))
+    vm = env.get_vm(params["main_vm"])
+    serials = params["serials"].split()
+    v_path = vm.get_serial_console_filename(serials[-1])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    out_put = session.cmd_output("nohup cat /proc/kallsyms > /dev/virtio-ports/vs2 2>&1 &")
+    time.sleep(10)
+    if session.cmd_status_output("date")[1] is None:
```
Comment (yanan-fu) [inline qemu/tests/virtio_trace_pipenb.py:31]: Why do not use `session.cmd_output` directly ?

File: qemu/tests/virtio_trace_pipenb.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        test.fail("There should be date in the output!")
```
Comment (yanan-fu) [inline qemu/tests/virtio_trace_pipenb.py:32]: Better explain no date means guest freeze after write data to the virtio-serial port while there's no reader process in the host.

**Comment by nanliu-r** [comment]
Test pass:
Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.virtio_trace_pipenb.q35: PASS (72.99 s)

File: qemu/tests/virtio_trace_pipenb.py (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/virtio_trace_pipenb.py]: update, please help to check.

File: qemu/tests/virtio_trace_pipenb.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/virtio_trace_pipenb.py]: update and thanks.

Comment (nanliu-r) [reply qemu/tests/virtio_trace_pipenb.py]: update and thanks.

**Comment by nanliu-r** [comment]
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.virtio_trace_pipenb.q35: PASS (75.05 s)

**Comment by yanan-fu** [comment]
LGTM

Acked-by: Yanan Fu <yfu@redhat.com>

**Pull request merged**